### PR TITLE
add TSC and PCS rad form factor shapes

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request extends the currently existing CIC particle shape (for radiation) for the TSC and PCS shape. 

**Requires:**
 - [x] merge of #1686 
 - [x] test CIC, TSC and PCS shape again
 - [x] add documentation on new shapes in `radiationConfig.param`

### CODE REVIEW

File: src/picongpu/include/plugins/radiation/radFormFactor.hpp [hunk 1]
```diff
@@ -44,17 +44,18 @@ namespace picongpu
        */
       HDINLINE float_X operator()(const float_X N, const float_X omega, const vector_X observer_unit_vec) const
       {
+        //util::Pow<float_X, 1> pow;
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/radiation/radFormFactor.hpp]: please remove this comment or dead code

File: src/picongpu/include/plugins/radiation/radFormFactor.hpp [hunk 2]
```diff
@@ -44,17 +44,18 @@ namespace picongpu
...
+        const float_X sincX = util::pow( math::sinc( observer_unit_vec.x() * CELL_WIDTH / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ), 1 );
+        const float_X sincY = util::pow( math::sinc( observer_unit_vec.y() * CELL_HEIGHT / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ), 1 );
+        const float_X sincZ = util::pow( math::sinc( observer_unit_vec.z() * CELL_DEPTH / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ), 1 );
           return math::sqrt(
-              N + ( N * N - N ) * util::square(
-                  math::sinc( observer_unit_vec.x() * CELL_WIDTH / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ) *
-                  math::sinc( observer_unit_vec.y() * CELL_HEIGHT / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ) *
-                  math::sinc( observer_unit_vec.z() * CELL_DEPTH / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega )
-              ) // util::square
+              N + ( N * N - N ) * util::square( sincX * sincY * sincZ )
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/radiation/radFormFactor.hpp]: only a note: `util::square` == `util::pow`

File: src/picongpu/include/plugins/radiation/radFormFactor.hpp [hunk 3]
```diff
@@ -44,17 +44,18 @@ namespace picongpu
        */
       HDINLINE float_X operator()(const float_X N, const float_X omega, const vector_X observer_unit_vec) const
       {
+        //util::Pow<float_X, 1> pow;
+        const float_X sincX = util::pow( math::sinc( observer_unit_vec.x() * CELL_WIDTH / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ), 1 );
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/radiation/radFormFactor.hpp]: Why do you use pow(X,1) here?

Comment (psychocoderHPC) [inline src/picongpu/include/plugins/radiation/radFormFactor.hpp]: If the shapes only differ in the used pow function than please show it in your code:
```C++
template< uint32_t T_order >
struct BaseShape
{
   HDINLINE float_X operator()(const float_X N, const float_X omega, const vector_X observer_unit_vec) const
  {
     //....
     // pow(T_order);
  }
};

struct radFormFactor_TSC_3D : public BaseShape< 2 >
{

};
```

Maybe a function parameter with the power factor is also possible.

File: src/picongpu/include/plugins/radiation/radFormFactor.hpp [hunk 4]
```diff
@@ -84,6 +85,69 @@ namespace picongpu
...
+       *
+       * @param N = macro particle weighting
+       * @param omega = frequency at which to calculate the  form factor
+       * @param observer_unit_vec = observation direction
+       * @return the Form Factor: sqrt( | \mathcal{F} |^2 )
+       */
+      HDINLINE float_X operator()(const float_X N, const float_X omega, const vector_X observer_unit_vec) const
+      {
+        //util::Pow<float_X, 2> pow;
+        const float_X sincX = util::pow( math::sinc( observer_unit_vec.x() * CELL_WIDTH / ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ), 2 );
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/radiation/radFormFactor.hpp]: also possible to write it like this:

```C++
float3_X sincValue;
for( uint32_t d = 0; d < DIM3; ++d)
    sincValue[d] = util::pow( 
        math::sinc( observer_unit_vec.y() * cellSize[d] / 
            ( SPEED_OF_LIGHT * float_X( 2.0 ) ) * omega ), 
        2 
    );

return math::sqrt(
    N + ( N * N - N ) * util::square(  sincValue::productOfComponents()  
)

```

File: src/picongpu/include/plugins/radiation/radFormFactor.hpp (hunk 2 above)
Comment (ax3l) [inline src/picongpu/include/plugins/radiation/radFormFactor.hpp]: note: squares (^N) can be implemented more efficiently than the more general pow (^N)
note2: but with the `pow` implementation right now you are right

**Comment by PrometheusPi** [comment]
@psychocoderHPC let me know when I can squash the last two commits

**Comment by PrometheusPi** [comment]
@psychocoderHPC - rebased after white space travis check inclusion

**Comment by PrometheusPi** [comment]
@psychocoderHPC Any more changes needed? May a squash the commits?

**Comment by psychocoderHPC** [comment]
Looks good from my side

**Pull request merged**